tests/system/test_aggregation_query.py

```diff
@@ -70,54 +70,299 @@ def nested_query(aggregation_query_client, ancestor_key):
 
 
 @pytest.mark.parametrize("database_id", [None, _helpers.TEST_DATABASE], indirect=True)
-def test_aggregation_query_default(aggregation_query_client, nested_query, database_id):
+def test_count_query_default(aggregation_query_client, nested_query, database_id):
     query = nested_query
 
     aggregation_query = aggregation_query_client.aggregation_query(query)
     aggregation_query.count()
     result = _do_fetch(aggregation_query)
     assert len(result) == 1
-    for r in result[0]:
-        assert r.alias == "property_1"
-        assert r.value == 8
+    assert len(result[0]) == 1
+    r = result[0][0]
+    assert r.alias == "property_1"
+    expected_count = len(populate_datastore.CHARACTERS)
+    assert r.value == expected_count
 
 
 @pytest.mark.parametrize("database_id", [None, _helpers.TEST_DATABASE], indirect=True)
-def test_aggregation_query_with_alias(
-    aggregation_query_client, nested_query, database_id
+@pytest.mark.parametrize(
+    "aggregation_type,aggregation_args,expected",
+    [
+        ("count", (), len(populate_datastore.CHARACTERS)),
+        (
+            "sum",
+            ("appearances",),
+            sum(c["appearances"] for c in populate_datastore.CHARACTERS),
+        ),
+        (
+            "avg",
+            ("appearances",),
+            sum(c["appearances"] for c in populate_datastore.CHARACTERS)
+            / len(populate_datastore.CHARACTERS),
+        ),
+    ],
+)
+def test_aggregation_query_in_transaction(
+    aggregation_query_client,
+    nested_query,
+    database_id,
+    aggregation_type,
+    aggregation_args,
+    expected,
 ):
+    """
+    When an aggregation query is run in a transaction, the transaction id should be sent with the request.
+    The result is the same as when it is run outside of a transaction.
+    """
+    import mock
```

Comment:
linked above, maybe we can make these more in line with https://github.com/googleapis/python-datastore/blob/main/tests/system/test_transaction.py so that we don't have to do any mocking

Reply:
The mock here is used to capture the transaction field sent along with the request, which I don't believe is captured by the other tests. 

But yeah, on second thought that part probably doesn't belong here. I broke this test up and moved those parts into the unit tests